cmd/gmnhg/templates.go

```diff
@@ -44,8 +44,35 @@ var defaultSingleTemplate = mustParseTmpl("single", `# {{ .Metadata.PostTitle }}
 var defaultIndexTemplate = mustParseTmpl("index", `# Site index
 
 {{ with .Content }}{{ printf "%s" . -}}{{ end }}
-{{ range $dir, $posts := .PostData }}Index of {{ $dir }}:
+{{ range $dir, $posts := .PostData }}{{ if $dir }}Index of {{ $dir }}:
 
 {{ range $p := $posts | sortPosts }}=> {{ $p.Link }} {{ $p.Metadata.PostDate.Format "2006-01-02 15:04" }} - {{ $p.Metadata.PostTitle }}
-{{ end }}{{ end }}
+{{ end }}{{ end }}{{ end }}
+`)
+
+var defaultRssTemplate = mustParseTmpl("rss", `{{- $Site := .Site -}}
+<?xml version="1.0" encoding="utf-8" standalone="yes"?>
+<rss version="2.0" xmlns:atom="http://www.w3.org/2005/Atom">
+  <channel>
+    <title>{{ if $Site.Title }}{{ $Site.Title }}{{ else }}Site feed{{ with .Dirname }} for {{.}}{{end}}{{end}}</title>
+    <link>{{ $Site.GeminiBaseURL }}{{ .Link }}</link>
+    <description>Recent content{{ with .Dirname }} in {{.}}{{end}}{{ with $Site.Title }} on {{.}}{{end}}</description>
+    <generator>gmnhg</generator>{{ with $Site.LanguageCode }}
+    <language>{{.}}</language>{{end}}{{ with $Site.Author.email }}
+    <managingEditor>{{.}}{{ with $Site.Author.name }} ({{.}}){{end}}</managingEditor>
+    <webMaster>{{.}}{{ with $Site.Author.name }} ({{.}}){{end}}</webMaster>{{end}}{{ with $Site.Copyright }}
+    <copyright>{{.}}</copyright>{{end}}
+    <lastBuildDate>{{ now.Format "Mon, 02 Jan 2006 15:04:05 -0700" }}</lastBuildDate>
+    {{ printf "<atom:link href=%q rel=\"self\" type=\"application/rss+xml\" />" .Link }}
```

Comment:
We currently use `text/template`, which doesn't perform HTML/XML sanitization on itself. Is this safe to escape URIs with just the `%q` verb from `fmt`? `html/template` does this by adding `urlescape` and `htmlescape` filters.

Reply:
Good catch, I should escape that.